### ISSUE TITLE
Compiler API: A reference to an enum with one value references the enum value

### ISSUE THREAD

**Issue by Gerrit0** [microsoft/TypeScript#46755]
# Bug Report

### 🔎 Search Terms

single member enum, reference types, compiler API

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about symbol references

### 💻 Code

```ts
export enum CoolEnum {
//          ^^^^^^^^ symbol.id = 103
   FOO, 
// ^^^ symbol.id = 104
} 
 
export interface Response { 
    x:  CoolEnum 
//  ^ type.symbol.id = 104
}
```

If I add a second member to the enum, the symbol references change to work as expected.

```ts
export enum CoolEnum {
//          ^^^^^^^^ symbol.id = 103
    FOO,
//  ^^^ symbol.id = 104
    BAR,
//  ^^^ symbol.id = 105
} 
 
export interface Response { 
    x:  CoolEnum
//  ^ type.symbol.id = 103
}
```

### 🙁 Actual behavior

When enums have a single value, reference types which point to them have the type's symbol set to the only member of the enum, not to the enum's symbol.

### 🙂 Expected behavior

When enums have any number of values, reference types pointing to them should point to the enum itself.

**Comment by fatcerberus** [comment]
Not familiar with the compiler codebase, but my guess is this happens because an enum type is equivalent to the union of its members, similar to how `boolean` is the same type as `true | false`.  So if there's only one member then `CoolEnum` and `CoolEnum.Foo` are in fact the same type.

**Comment by andrewbranch** [comment]
That’s correct

**Comment by andrewbranch** [comment]
That said, we do construct two separate copies of the enum type with value `1` in the single member case, one returned for `getTypeOfSymbol` for the enum declaration symbol and one for the member declaration symbol. This is a little weird.

**Comment by andrewbranch** [comment]
@Gerrit0 you can try a draft PR that changes this; sometimes we just don’t have great coverage/visibility about what the public checker API returns. If it doesn’t cause any breaks/changes internally we might consider it, but no guarantees.

**Comment by Gerrit0** [comment]
> That said, we do construct two separate copies of the enum type with value 1 in the single member case

Yep... and this is why I think this is probably unintentional/buggy. I might have time to look into this.... in the meantime, how should I go from a `ReferenceType` to the declaration that the user intended? Symbols seem like they ought to be the way to do this.

**Comment by adamviola** [comment]
Hey there! Just wondering if there’s any update on this?

Found a few examples of this bug affecting docs in the wild:

Typedoc'd `vscode.d.ts`
- [`SymbolTag` enum has a single member: `Deprecated`]()
- [`CallHierarchyItem.tags` is documented to have type`Deprecated[]`; expected `SymbolTag[]`](https://vscode-api.js.org/classes/vscode.CallHierarchyItem.html)
- Happens a few more times in `vscode.d.ts`

Adobe PDF Services SDK
- [`NotifierType` enum has a single member: `CALLBACK`](https://opensource.adobe.com/pdfservices-node-sdk-samples/apidocs/latest/enums/NotifierType.html)
- [Constructor of `NotifierConfig` is documented to include parameter of type `CALLBACK`; expected `NotifierType`](https://opensource.adobe.com/pdfservices-node-sdk-samples/apidocs/latest/classes/NotifierConfig.html)

Sendbird Chat SDK
- [`MessageCollectionInitPolicy` has a single member `CACHE_AND_REPLACE_BY_API `](https://docs.sendbird.com/docs/chat/sdk/v4/javascript/ref/enums/_sendbird_chat_groupChannel.MessageCollectionInitPolicy.html)
- [`MessageCollection.initialize`'s `policy` parameter is documented to have type `CACHE_AND_REPLACE_BY_API`; expected `MessageCollectionInitPolicy`](https://docs.sendbird.com/docs/chat/sdk/v4/javascript/ref/classes/_sendbird_chat_groupChannel.MessageCollection.html#initialize)
